core/src/autogluon/core/hpo/executors.py

```diff
@@ -0,0 +1,370 @@
+        )
+        if 'num_trials' not in hyperparameter_tune_kwargs and default_num_trials is not None:
+            hyperparameter_tune_kwargs['num_trials'] = default_num_trials
+        self.hyperparameter_tune_kwargs = hyperparameter_tune_kwargs
+        return time_limit
+    
+    def register_resources(self, initialized_model_base):
+        user_cpu_count = initialized_model_base._get_child_aux_val(key='num_cpus', default=None)
+        user_gpu_count = initialized_model_base._get_child_aux_val(key='num_gpus', default=None)
+        resource_kwargs = initialized_model_base._preprocess_fit_resources()
+
+        num_gpus = ResourceCalculator.get_total_gpu_count(
+            user_specified_num_gpus=user_gpu_count,
+            model_default_num_gpus=resource_kwargs.get('num_gpus', 0),
+        )
+        num_cpus = ResourceCalculator.get_total_cpu_count(
+            user_specified_num_cpus=user_cpu_count,
+            model_default_num_cpus=resource_kwargs.get('num_cpus', 0),
+        )
+        
+        self.resources = dict(num_gpus=num_gpus, num_cpus=num_cpus)
+        
+    def validate_search_space(self, search_space, model_name):
+        from ray.tune.sample import Domain
+        if not any(isinstance(search_space[hyperparam], (Space, Domain)) for hyperparam in search_space):
+            logger.warning(f"\tNo hyperparameter search space specified for {model_name}. Skipping HPO. "
+                           f"Will train one model based on the provided hyperparameters.")
+            raise EmptySearchSpace
+        self.search_space = search_space
+        logger.log(15, f"\tHyperparameter search space for {model_name}: ")
+        for hyperparam in search_space:
+                if isinstance(search_space[hyperparam], (Space, Domain)):
+                    logger.log(15, f"{hyperparam}:   {search_space[hyperparam]}")
+                    
+    def execute(
+            self,
+            model_trial,
+            train_fn_kwargs,
+            directory,
+            minimum_cpu_per_trial,
+            minimum_gpu_per_trial,
+            model_estimate_memory_usage,
+            adapter_type,
+        ):
+        from .ray_hpo import (
+            run,
+            RayTuneAdapterFactory
+        )
+        analysis = run(
+            trainable=model_trial,
+            trainable_args=train_fn_kwargs,
+            search_space=self.search_space,
+            hyperparameter_tune_kwargs=self.hyperparameter_tune_kwargs,
+            metric='validation_performance',
+            mode='max',
+            save_dir=directory,
+            ray_tune_adapter=RayTuneAdapterFactory.get_adapter(adapter_type)(),
+            total_resources=self.resources,
+            minimum_cpu_per_trial=minimum_cpu_per_trial,
+            minimum_gpu_per_trial=minimum_gpu_per_trial,
+            model_estimate_memory_usage=model_estimate_memory_usage,
+            time_budget_s=self.time_limit,
+            verbose=0,
```

Comment:
why verbose hard-coded to 0?

Reply:
The log of ray hpo looks not good with autogluon, so I just make it 0.